sample_rock/sample-app/app.py

```diff
@@ -1,10 +1,19 @@
 # Copyright 2023 Canonical Ltd.
 # See LICENSE file for licensing details.
+import time
 
-from flask import Flask
+from flask import Flask, request
 
 app = Flask(__name__)
 
+
 @app.route("/")
 def hello_world():
     return "Hello, World!"
+
+
+@app.route("/sleep")
+def sleep():
+    duration_seconds = int(request.args.get("duration"))
+    time.sleep(duration_seconds)
+    return ""
```

Comment:
It this return value necessary?

Reply:
Yes, the return value of a flask view function can not be `None`.

Reply:
Perhaps better to return HTTP 204
`return "", http.HTTPStatus.NO_CONTENT`